astropy/io/fits/fitstime.py

```diff
@@ -0,0 +1,251 @@
+                        'JDREF' : (0.0,'Time columns are jd = jd1 + jd2'),
+                        'TREFPOS' : ('TOPOCENTER','Time reference position')}
+
+    # Compatibility of Time Scales and Reference Positions
+    TIME_SCALE_REF = {'tai': 'TOPOCENTER',
+                      'tt' : 'TOPOCENTER',
+                      'ut1' : 'TOPOCENTER',
+                      'utc' : 'TOPOCENTER',
+                      'tcg' : 'GEOCENTER',
+                      'tcb' : 'BARYCENTER',
+                      'tdb' : 'BARYCENTER'}
+
+    def __init__(self):
+        # Set defaults for global time scale, reference, etc.
+        self.global_info = {'scale' : 'UTC',
+                            'mjdref' : None,
+                            'unit' : 's'}
+        # Set default dictionary for time columns
+        self.time_columns = defaultdict(OrderedDict)
+        self.astropy_file = None
+
+    @classmethod
+    def is_time_column_keyword(cls, keyword):
+        """
+        Check if the FITS header keyword is a time column-specific keyword.
+
+        Parameters
+        ----------
+        keyword : str
+            FITS keyword.
+        """
+        if keyword[5:].isdigit() and keyword[:5] in cls.COLUMN_TIME_KEYWORDS:
+            return True
+        return False
+
+    def set_global_time(self, key, value, comment=None):
+        """
+        Set the global time reference frame attributes.
+
+        Parameters
+        ----------
+        key : str
+            FITS global time reference frame keyword.
+        value : int, str, list
+            value associated with specified keyword.
+        comment : str
+            comment associated with specified keyword.
+        """
+        if key in self.TIME_KEYWORDS:
+            self.global_info[self.TIME_KEYWORDS[key]] = value
+        else:
+            raise ValueError('Illegal global time keyword.')
+
+    def set_column_override(self, key, value, comment=None):
+        """
+        Set the time column specific override attributes.
+
+        Parameters
+        ----------
+        key : str
+            FITS time column specific keyword.
+        value : int, str, list
+            value associated with specified keyword.
+        comment : str
+            comment associated with specified keyword.
+        """
+        if self.is_time_column_keyword(key):
+            idx = int(key[5:])
+            self.time_columns[idx][self.COLUMN_TIME_KEYWORDS[key[:5]]] = value
+        else:
+            raise ValueError('Illegal column-specific time keyword.')
+
+    def convert_time_columns(self, table):
+        """
+        Convert time columns to Astropy Time columns.
+
+        Parameters
+        ----------
+        table : astropy.table.Table
+            The table whose time columns are to be converted.
+        """
+        for idx, time_col in self.time_columns.items():
+            time_colname = table.colnames[idx - 1]
+            if self.astropy_file is not None:
```

Comment:
Get rid of this.  Eventually there may be optional astropy-specific header keywords (e.g. providing the format), but the handling can be based purely on what keywords / data are found in the file, not a specific marker.